src/main/java/com/synopsys/integration/alert/web/controller/UIComponentController.java

```diff
@@ -72,48 +72,48 @@ public UIComponentController(final DescriptorMap descriptorMap) {
                     final ConfigContextEnum contextType = EnumUtils.getEnum(ConfigContextEnum.class, context);
                     final UIConfig uiConfig = descriptor.getUIConfig(contextType);
                     if (uiConfig != null) {
-                        return Arrays.asList(uiConfig.generateUIComponent());
+                        return Arrays.asList(uiConfig.generateDescriptorMetadata());
                     } else {
                         return Collections.emptyList();
                     }
                 } else {
                     // name only
-                    return descriptor.getAllUIConfigs().stream().map(UIConfig::generateUIComponent).collect(Collectors.toList());
+                    return descriptor.getAllUIConfigs().stream().map(UIConfig::generateDescriptorMetadata).collect(Collectors.toList());
                 }
             } else {
                 return Collections.emptyList();
             }
         } else if (StringUtils.isNotBlank(context)) {
             final ConfigContextEnum contextType = EnumUtils.getEnum(ConfigContextEnum.class, context);
-            return descriptorMap.getUIComponents(contextType);
+            return descriptorMap.getDescriptorMetadataList(contextType);
         } else {
             return descriptorMap.getAllUIComponents();
         }
     }
 
     // TODO split out provider and distribution endpoints
     @GetMapping("/distribution")
-    public UIComponent getDistributionUIComponent(@RequestParam(value = "providerName", required = true) final String providerName, @RequestParam(value = "channelName", required = true) final String channelName) {
+    public DescriptorMetadata getDistributionUIComponent(@RequestParam(value = "providerName", required = true) final String providerName, @RequestParam(value = "channelName", required = true) final String channelName) {
```

Comment:
I believe there will be a time where you hit this endpoint and only know the channel name and not the provider name. So we might want to support that scenario.

Reply:
I don't know if we will use this class going forward. We might only want to expose metadata in terms of descriptors.